prymer/primer3/primer3.py

```diff
@@ -371,12 +366,16 @@ def design_primers(self, design_input: Primer3Input) -> Primer3Result:  # noqa:
                 f"Error, trying to use a subprocess that has already been "
                 f"terminated, return code {self._subprocess.returncode}"
             )
-
-        design_region: Span = self._create_design_region(
-            target_region=design_input.target,
-            max_amplicon_length=design_input.primer_and_amplicon_params.max_amplicon_length,
-            min_primer_length=design_input.primer_and_amplicon_params.min_primer_length,
-        )
+        design_region: Span
+        match design_input.task:
+            case PickHybProbeOnly():
+                design_region = design_input.target
+            case _:
```

Comment:
**suggestion** consider requiring that the provided task is one of the three design tasks before executing this block, and keep the `case _` branch to raise an error for any other task